src/index.ts

```diff
@@ -597,9 +589,95 @@ export async function precomputedInit(
     throwOnFailedInitialization: true, // always use true here as underlying instance fetch is surrounded by try/catch
     skipInitialPoll: skipInitialRequest,
   };
-  instance.setSubjectAndPrecomputedFlagsRequestParameters(precomputedFlagsRequestParameters);
 
-  await instance.fetchPrecomputedFlags();
+  const subject: Subject = { subjectKey, subjectAttributes };
+
+  EppoPrecomputedJSClient.instance = new EppoPrecomputedJSClient({
+    precomputedFlagStore: memoryOnlyPrecomputedFlagsStore,
+    requestParameters,
+    subject,
+  });
+
+  EppoPrecomputedJSClient.instance.setAssignmentLogger(config.assignmentLogger);
+  await EppoPrecomputedJSClient.instance.fetchPrecomputedFlags();
+
+  EppoPrecomputedJSClient.initialized = true;
+  return EppoPrecomputedJSClient.instance;
+}
+
+/**
+ * Initializes the Eppo precomputed client with configuration parameters.
+ *
+ * The purpose is for use-cases where the precomputed assignments are available from an external process
+ * that can bootstrap the SDK.
+ *
+ * This method should be called once on application startup.
+ *
+ * @param config - client configuration
+ * @returns a singleton precomputed client instance
+ * @public
+ */
+export interface IPrecomputedClientConfigSync {
+  precomputedConfigurationWire: string;
+  assignmentLogger?: IAssignmentLogger;
+  throwOnFailedInitialization?: boolean;
+}
+
+/**
+ * Initializes the Eppo precomputed client with configuration parameters.
+ *
+ * The purpose is for use-cases where the precomputed assignments are available from an external process
+ * that can bootstrap the SDK.
+ *
+ * This method should be called once on application startup.
+ *
+ * @param config - precomputed client configuration
+ * @returns a singleton precomputed client instance
+ * @public
+ */
+export function offlinePrecomputedInit(
+  config: IPrecomputedClientConfigSync,
+): EppoPrecomputedClient {
+  const throwOnFailedInitialization = config.throwOnFailedInitialization ?? true;
+
+  const configurationWire: IConfigurationWire = JSON.parse(config.precomputedConfigurationWire);
+  if (!configurationWire.precomputed) {
+    applicationLogger.error('Invalid precomputed configuration wire');
+    return EppoPrecomputedJSClient.instance;
+  }
+  const { subjectKey, subjectAttributes, response } = configurationWire.precomputed;
+  const parsedResponse: IPrecomputedConfigurationResponse = JSON.parse(response);
+
+  try {
+    const memoryOnlyPrecomputedStore = precomputedFlagsStorageFactory();
+    memoryOnlyPrecomputedStore
+      .setEntries(parsedResponse.flags)
+      .catch((err) =>
+        applicationLogger.warn('Error setting precomputed assignments for memory-only store', err),
+      );
+    memoryOnlyPrecomputedStore.salt = parsedResponse.salt;
+
+    const subject: Subject = {
+      subjectKey,
+      subjectAttributes: subjectAttributes ?? ({} as Attributes),
+    };
+
+    EppoPrecomputedJSClient.instance = new EppoPrecomputedJSClient({
```

Comment:
major: before potentially overwriting previous instance, we should shut it down. (and warn of re-initialization)

Reply:
I think I'm unclear about what "shut it down" means here. The client is offline so polling/making requests isn't at play. Also the two instance variables being set by the constructor (`subject` and `precomputedFlagStore`) are private. 

I could make these public so I can clear them here. Or I can add a public method to `EppoPrecomputedClient` in the common sdk to clear these. I can't do it in the constructor because that only has access to the new instance, not the one being replaced.

Reply:
My attempt at a private shutdown function

```
function shutdownEppoPrecomputedClient() {
  if (EppoPrecomputedJSClient.instance) {
    EppoPrecomputedJSClient.instance.stopPolling();
    EppoPrecomputedJSClient.initialized = false;
    applicationLogger.warn(`${loggerPrefix} Precomputed client is being re-initialized.`);
  }
}
```

The configuration store is a private variable in `EppoPrecomputedClient` so I can't clear it here, not does it come with any helpers to clear its contents. Also the store we use at the moment is memory-only, so I don't know if persisting across sessions is possible, e.g. if a page refresh is the reason for re-initializing

Reply:
> I think I'm unclear about what "shut it down" means here.

I mostly meant stopping polling. Most SDKs have a method for stopping any activity.

> The client is offline so polling/making requests isn't at play. 

The fact that we're initializing offline doesn't mean that the previous client was in offline mode too